lib/internal/bootstrap/pre_execution.js

```diff
@@ -11,7 +11,15 @@ const { Buffer } = require('buffer');
 const { ERR_MANIFEST_ASSERT_INTEGRITY } = require('internal/errors').codes;
 const assert = require('internal/assert');
 
+// TODO(addaleax): Remove the flag guarding against a double run of this,
+// it's only needed because of lib/internal/bootstrap/environment.js
+// which should also be removed (because the legacy `CreateEnvironment()`
+// also runs this file in addition to `LoadEnvironment()`, and the former
+// shouldn't).
+let ranPrepareMainThreadExecution = false;
```

Comment:
i think this can be removed now that https://github.com/nodejs/node/pull/31910 has landed?

Reply:
Right, thanks for catching that :) Done!